module/product/src/Application/Validator/ProductAttributeValidator.php

```diff
@@ -50,32 +53,47 @@ public function validate($value, Constraint $constraint): void
             return;
         }
 
-        if (!$value instanceof UpdateAttributeValueFormModel) {
+        if (!$value instanceof UpdateProductAttributeFormModel) {
             return; //maybe Exception
         }
 
-        if ($value->id === null || !AttributeId::isValid($value->id)) {
+        if ($value->id === null || !ProductId::isValid($value->id)) {
             return;
         }
 
-        $attribute = $this->attributeRepository->load(new AttributeId($value->id));
-        if (null === $attribute) {
-            return;
-        }
-
-        $secondConstraint = $this->provider->provide($attribute);
         $i = 0;
-        foreach ($value->values as $valueRow) {
-            $violations = $this->validator->validate(['value' => $valueRow->value], $secondConstraint);
-            if (0 < $violations->count()) {
-                /** @var ConstraintViolation $violation */
-                foreach ($violations as $violation) {
-                    $this->context
-                        ->buildViolation($violation->getMessage())
-                        ->atPath('values['.$i.'].value')
-                        ->addViolation();
+        foreach ($value->payload as $payloadValues) {
+            if ($payloadValues->id === null || !AttributeId::isValid($payloadValues->id)) {
+                continue;
+            }
+            $attribute = $this->attributeRepository->load(new AttributeId($payloadValues->id));
+            if (null === $attribute) {
+                continue;
```

Comment:
Is attribute existence covered by other constraint? Since it's validation I'm not sure whether invalid attributes should be ignored - perhaps an constraint violation should be raised

Reply:
Attribute existence is validated here 
https://github.com/ergonode/backend/blob/b2a790d0e47acd7b4c427334c1e8f1b087cab598/module/product/src/Application/Model/Product/Attribute/Update/UpdateAttributeValueFormModel.php#L23